src/software/amazon/ion/IonReader.java

```diff
@@ -172,6 +172,17 @@
     public IonType getType();
 
     /**
+     * Returns the actual type of the Ion <code>int</code> at the current
+     * value.
+     *
+     * @return an {@link IntegerType} denoting the current <code>int</code>
+     * value's Java type, or <code>null</code> if the current value is
+     * <code>null.int</code>.
+     * @see IonInt#getIntegerType()
+     */
+    public IntegerType getIntegerType();
```

Comment:
This is underspecified; there's four relevant states and only two are documented:

  * The current value is an actual (i.e., non-null) `int`
  * The current valus is `null.int`
  * The current value is not an `int`
  * There is no current value

The latter two are not specified.

Following Almann's note, I'm not sure there's any value in distinguishing between any of the latter tree states, so I think just returning `null` is sufficient and `IntegerType.NONE` is not needed.